tools/clip/run.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python3
+
+from argparse import ArgumentParser, ArgumentTypeError
+from cereal.messaging import SubMaster
+from collections.abc import Sequence
+from openpilot.common.api import api_get
+from openpilot.common.prefix import OpenpilotPrefix
+from pathlib import Path
+from random import randint
+from subprocess import Popen, PIPE
+from typing import Literal
+import atexit
+import logging
+import os
+import platform
+import shutil
+import time
+
+DEFAULT_OUTPUT = 'output.mp4'
+DEMO_START = 90
+DEMO_END = 105
+DEMO_ROUTE = 'a2a0ccea32023010/2023-07-27--13-01-19'
+FRAMERATE = 20
+PIXEL_DEPTH = '24'
+RESOLUTION = '2160x1080'
+SECONDS_TO_WARM = 2
+PROC_WAIT_SECONDS = 5
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+REPLAY = str(Path(SCRIPT_DIR, '../../tools/replay/replay').resolve())
+UI = str(Path(SCRIPT_DIR, '../../selfdrive/ui/ui').resolve())
+
+logger = logging.getLogger('clip.py')
+
+
+def check_for_failure(proc: Popen):
+  exit_code = proc.poll()
+  if exit_code is not None and exit_code != 0:
+    cmd = str(proc.args)
+    if isinstance(proc.args, str):
+      cmd = proc.args
+    elif isinstance(proc.args, Sequence):
+      cmd = str(proc.args[0])
+    msg = f'{cmd} failed, exit code {exit_code}'
+    logger.error(msg)
+    stdout, stderr = proc.communicate()
+    if stdout:
+      logger.error(stdout.decode())
+    if stderr:
+      logger.error(stderr.decode())
+    raise ChildProcessError(msg)
+
+
+def get_route(route: str):
```

Comment:
`tools.lib.route.Route` should have everything you need. if it's not there, we should add it